### ISSUE TITLE
Dark theme for iOS

### ISSUE THREAD

**Issue by iadcialim** [JetBrains/compose-multiplatform#3213]
How to implement dark theming for iOS app target?

**Comment by dima-avdeev-jb** [comment]
Thanks, good question. We will prepare a tutorial for this topic later.

**Comment by dima-avdeev-jb** [comment]
For now you can already use
```Kotlin
MaterialTheme(colors = if(isSystemInDarkTheme()) darkColors() else lightColors(), ) {
// ...
```

**Comment by iadcialim** [comment]
> ```kotlin
> isSystemInDarkTheme
> ```

`isSystemInDarkTheme` does not work for iOS for me and only for android.
Is there something to aside from that for iOS to work?

**Comment by dima-avdeev-jb** [comment]
isSystemInDarkTheme currently gets value only once after the application starts and not changes on runtime.
We will fix it

**Comment by zacharee** [comment]
iOS also has an issue where rotating the device will cause the background to flash white. Supposedly it's a bug with SwiftUI, although none of the solutions I found for SwiftUI worked for me. Is there a workaround we can use in Compose?

**Comment by dima-avdeev-jb** [comment]
> rotating the device will cause the background to flash white
Yes, we know it and will fix it in the future. But for now we are focusing on other issues

> Is there a workaround we can use in Compose?
I don't know the proper workaround for it. Better to wait for our fix.

**Comment by Sanlorng** [comment]
> > rotating the device will cause the background to flash white
> 
> Yes, we know it and will fix it in the future. But for now we are focusing on other issues
> 
> > Is there a workaround we can use in Compose?
> 
> I don't know the proper workaround for it. Better to wait for our fix.
> 

The `isSystemDarkTheme` changes at runtime will also land on the Windows JVM target?

**Comment by Sanlorng** [comment]
> > rotating the device will cause the background to flash white
> 
> Yes, we know it and will fix it in the future. But for now we are focusing on other issues
> 
> > Is there a workaround we can use in Compose?
> 
> I don't know the proper workaround for it. Better to wait for our fix.
> 

The `isSystemDarkTheme` changes at runtime will also land on the Windows JVM target?

**Comment by Alex009** [comment]
maybe useful. i implement runtime observer of dark mode changes for ios:
```kotlin
@Composable
actual fun isSystemInDarkTheme(): Boolean {
    var style: UIUserInterfaceStyle by remember {
        mutableStateOf(UITraitCollection.currentTraitCollection.userInterfaceStyle)
    }

    val viewController: UIViewController = LocalUIViewController.current
    DisposableEffect(Unit) {
        val view: UIView = viewController.view
        val traitView = TraitView {
            style = UITraitCollection.currentTraitCollection.userInterfaceStyle
        }
        view.addSubview(traitView)

        onDispose {
            traitView.removeFromSuperview()
        }
    }

    return style == UIUserInterfaceStyle.UIUserInterfaceStyleDark
}

@ExportObjCClass
private class TraitView(
    private val onTraitChanged: () -> Unit
) : UIView(frame = CGRectZero.readValue()) {
    override fun traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
        onTraitChanged()
    }
}
```

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3213">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by delacrixmorgan** [comment]
If you're using the `androidx.compose.foundation.isSystemInDarkTheme()`, then it should be working with `1.6.10`.

Problem with what I faced was that I forgot to initiate it in my `MainViewController` where our `AppTheme` is initialised. So, make sure your `AppTheme` implementation in the `MainViewController`, where your `ComposeUIViewController` lives is configured properly.
